### PR TITLE
Remove readDataSetToEnd

### PR DESCRIPTION
**Pull request by maltelueken**
Removes the deprecated `readDataSetToEnd` function.

Fixes https://github.com/jasp-stats/jasp-test-release/issues/2784
Fixes https://github.com/jasp-stats/jasp-test-release/issues/2783
Fixes https://github.com/jasp-stats/jasp-issues/issues/2942
Closes https://github.com/jasp-stats/jasp-issues/issues/3019

### CODE REVIEW

**Comment by maltelueken** [comment]
Some tests are failing, but these were failing before.

**Comment by maltelueken** [comment]
@JohnnyDoorn or @tomtomme, could you check this PR?

File: R/classicProcess.R
Comment (shun2wang) [inline R/classicProcess.R:471]: Hi, here seems still legacy code...

Comment (maltelueken) [reply R/classicProcess.R]: Thanks! Removed this function.

**Comment by shun2wang** [comment]
[ModerationBugAlbumSales_2.jasp.zip](https://github.com/user-attachments/files/17949867/ModerationBugAlbumSales_2.jasp.zip)
Hi @maltelueken  I tried this PR and it have other problems with this file (remove `.zip`) , thanks.

**Comment by maltelueken** [comment]
It seems like the QML constraints are broken (usually an error message appears if you select the same variable twice in one row). I don't know how to fix this since the QML code should still be correct. @boutinb, do you have an idea what could be wrong?

**Comment by shun2wang** [comment]
>It seems like the QML constraints are broken...

OK then we can leave it to fix after release because it just a bad/accident behavior of "user" side.

**Review by shun2wang — APPROVED** [review]

**Comment by boutinb** [comment]
I have added a commit to fix some problems with the width of the dropdowns in the Input Variables.
Be careful: this will work only with the newest Nightly (it is now building): the fixedWidth property was not settable until now.
In this commit I also change the allowedColumns in the "Mediators M" VariablesList of the InputModelNumber from `["scale", "ordinal"] `to `["scale", "nominal"]`, because if I added a variable from the Categorical Predictors (which is Nominal), it transformed into a scale because it was neither a scale or an ordinal.


But when testing I saw other problems: I have tested the data sent by @shun2wang, and got this error:
Screenshot 2024-12-02 at 17 12 02

**Comment by maltelueken** [comment]
Thanks! I cannot reproduce your error, can you send the JASP file?

**Comment by boutinb** [comment]
> Thanks! I cannot reproduce your error, can you send the JASP file?

I see that there is another problem when reading a JASP file with such an analysis. I'll have a look.
But the problem I described can be found by loading this csv file:

- add a Classical Process Model
- add adverts to the Dependent Variable
- add sales and airplay to Continuous Predictors
- in Model 1, set From: sales, To: adverts, Process Type: Mediator, Process Variable: sales.

Then you get the analysis error.

[ModerationBugAlbumSales.csv](https://github.com/user-attachments/files/17989964/ModerationBugAlbumSales.csv)

**Comment by shun2wang** [comment]
With Bruno's commit and latest build I found the "model" input fields was showing blank if you load that .jasp file above. click it has noting to do... @boutinb do you notice that?

**Comment by boutinb** [comment]
> With Bruno's commit and latest build I found the "model" input fields was showing blank in "Classic Process Model" after you load that .jasp file above. click it has noting to do... @boutinb do you notice that?

Yes....

**Comment by boutinb** [comment]
It took some time, but I have made a PR that will solve the QML issue: https://github.com/jasp-stats/jasp-desktop/pull/5753
When it is implemented, you can try again.

**Comment by shun2wang** [comment]
OK, I tried this with that PR and now QML problems seems solved. an R error still live.

**Comment by boutinb** [comment]
The last development in the Nightly has my PR https://static.jasp-stats.org/Nightlies/

**Comment by maltelueken** [comment]
When I use the nighly from 6-12 (`2afd5f9`), I get the following error:

image

In your screenshot, you used the one variable twice in the same row. Previously, QML would return an error in this case and the analysis would not run (which is why this error occurs). Is it possible to fix the QML to return this error again? The code to do that hasn't changed but it does not seem to work anymore.

**Comment by shun2wang** [comment]
Yes, that shoudn't block this PR, I have made another issue to track it.

**Review by shun2wang — APPROVED** [review]

**Comment by boutinb** [comment]
> When I use the nighly from 6-12 (`2afd5f9`), I get the following error:
> 
> image
> 
> I deleted the cache and used the latest commit of this branch.
> 
> In your screenshot, you used the one variable twice in the same row. Previously, QML would return an error in this case and the analysis would not run (which is why this error occurs). Is it possible to fix the QML to return this error again? The code to do that hasn't changed but it does not seem to work anymore.

You did not have my last commit on this PR: I removed the controlMinWidth in QML

I will try to fix the row validation issue.

**Pull request merged**
